src/parser/mod.rs

```diff
@@ -2734,6 +2734,10 @@ impl<'a> Parser<'a> {
                     let regexp = self.parse_keyword(Keyword::REGEXP);
                     let rlike = self.parse_keyword(Keyword::RLIKE);
                     if regexp || rlike {
+                        if self.parse_keyword(Keyword::ANY) {
+                            self.prev_token();
+                            return self.expected("pattern after RLIKE or REGEXP", self.peek_token());
+                        }
```

Comment:
Ah not sure I followed this part, we don't seem to take any action for when parsing ANY is successful?

Reply:
The only point of this condition was to add a meaningful error message when parsing incorrect syntax, for example: `RLIKE ANY`. See the last check of the unit test. Can be removed, I debated this one.

Reply:
Removed in the latest commit.